packages/langium/src/grammar/langium-grammar.langium

```diff
@@ -74,6 +74,12 @@ ParserRule:
     (definesHiddenTokens?='hidden' '(' (hiddenTokens+=[AbstractRule:ID] (',' hiddenTokens+=[AbstractRule:ID])*)? ')')? ':'
     definition=Alternatives ';';
 
+InfixRule: 'infix' RuleNameAndParams 'on' call=RuleCall ':' operators=InfixRuleOperators ';';
```

Comment:
We should allow `infers` and `returns` as usual. If a declared type is given, it *must* have the properties `left`, `right`, and `operator` with respective types.

Reply:
Ah 👉 https://github.com/eclipse-langium/langium/pull/1836#issuecomment-2715732901 – 👍 for a separate PR, but let's not forget that.